salt/state.py

```diff
@@ -3397,7 +3397,7 @@ def _filter_matches(_match, _data, _opts):
         ext_matches = self._master_tops()
         for saltenv in ext_matches:
             top_file_matches = matches.get(saltenv, [])
-            if self.opts['master_tops_first']:
+            if self.opts.get('master_tops_first'):
```

Comment:
Right, it is a key error because it is not defined in salt.config

All that is needed here for master_tops_first to always be defined is to add it to the default opts dictionary.

https://github.com/saltstack/salt/blob/2017.7/salt/config/__init__.py#L1523

Reply:
sounds good